packages/web-components/src/image/image.ts

```diff
@@ -0,0 +1,101 @@
+import { attr, FASTElement } from '@microsoft/fast-element';
+import { BorderRadius, ImageFit, ImageShape } from './image.options.js';
+
+/**
+ * The base class used for constucting a fluent image custom element
+ * @public
+ */
+export class Image extends FASTElement {
+  /**
+   * Image description
+   *
+   * @public
+   * @remarks
+   * HTML attribute: alt.
+   */
+  @attr
+  public alt: string;
+  /**
+   * presentation
+   *
+   * @public
+   * @remarks
+   * HTML attribute: role="presentation"
+   */
+  @attr
+  public role: string;
+  /**
+   * Image source
+   *
+   * @public
+   * @remarks
+   * HTML attribute: src.
+   */
+  @attr
+  public src: string;
+
+  /**
+   * Image layout
+   *
+   * @public
+   * @remarks
+   * HTML attribute: block.
+   */
+  @attr({ mode: 'boolean' })
+  public block?: boolean;
+  /**
+   * Image border
+   *
+   * @public
+   * @remarks
+   * HTML attribute: border.
+   */
+  @attr({ mode: 'boolean' })
+  public border?: boolean;
+  /**
+   * Image margin
+   *
+   * @public
+   * @remarks
+   * HTML attribute: margin.
+   */
+  @attr({ mode: 'boolean' })
+  public margin?: boolean;
```

Comment:
Is this part of our public API or is it additive? Not seeing it on the Fluent v9 storybook

Reply:
Nope, this is one prop that's not present in the Fluent v9 react component. I will add a note to the docs file stating where this component differs from the Fluent v9. That and border-radius look like the only two extras.

Reply:
Do you want me to make adjustments here?

Reply:
Do we think `margin` is the best name for what this is doing? Sure it adds a 16px margin around the host but it seems like an odd name as its a boolean.

Reply:
@halerankin I don't see this on the v9 image component - am I missing it?

Reply:
Here's the Figma spec I was given prior to building it out: https://www.figma.com/file/05wt6TAsEmgsCVZfPrpcWx/Image?node-id=0%3A1&t=KZwLafEjmVs2M7AG-0

Reply:
Sure - I think we need clarity here...though I'd argue if the intent is to add margin to the component, this is 100% the wrong way given there is additional cost of creating an attribute, adding bindings and lifecycle, etc. We should just use CSS in this case.

Reply:
Here's the latest v9 - just not seeing margin as enumerated as a prop, so I don't think we should move forward with it here (both for that reason and the one above)

Reply:
So, do we just leave it up to the dev implementing this component to add the 16px margin when needed?

Reply:
Absolutely - I think my question would be, "Why is margin special as far as CSS goes where it gets an attribute and others do not?"

Similarly, I don't see where we're actually doing anything *with that*.

Reply:
Looking at the spec, I see a prescribed layout: optional margin of 16px. We have no way to strictly require this 16px value. We *could* by adding a CSS class in our styles that provides that 16px but outside that we have zero control over margin value.

Reply:
I just looked at the spec and I see where the confusion is - I'm going to maintain that I don't think that is intended as part of the component API as far as attributes or properties are concerned. In the case that it's added to the react API, I'd probably hold on my stance given that margin should not be treated as a CSS property in a JS API given that we can easily apply it via CSS with zero cost. Were we to set this in the class as an attribute, we would need to then map it to css _from_ the JS class which would either be a unique token (odd, expensive) or an inline style (odd, unintuitive). 

To my point above, I think we need to understand why margin deserves special treatment over something like "padding". Further, what if I need margin-inline or just margin-block? Is the only method I have to set this setting a singular value for top/right/left/bottom? If so, we're likely creating confusion because...are those not supported values? If they are supported, I likely set them via CSS, so we return again to why margin is specially set for this scenario.

Reply:
In the spirit of keeping things simple, I think omitting margin as a component class property is best. Leaving the margin to the web component consumer provides them with more flexibility.